File: crossecore-csharp/Java/Java/Java_/NamespaceAccessBase.cs
/* CrossEcore is a cross-platform modeling framework that generates C#, TypeScript, 
 * JavaScript, Swift code from Ecore models with embedded OCL (http://www.crossecore.org/).
 * The original Eclipse Modeling Framework is available at https://www.eclipse.org/modeling/emf/.
 * 
 * contributor: Simon Schwichtenberg
 */
 
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using oclstdlib;
using Ecore;
namespace Java_{
	public class NamespaceAccessBase 
	:ASTNodeImpl, NamespaceAccess
	{
		
		protected override EClass eStaticClass() {
			return Java_PackageImpl.Literals.NAMESPACEACCESS;
		}
		
		
		
		public override Object eGet(int featureID, bool resolve, bool coreType) {
			switch (featureID) {
				case Java_PackageImpl.NAMESPACEACCESS_COMMENTS:
					return comments;
				case Java_PackageImpl.NAMESPACEACCESS_ORIGINALCOMPILATIONUNIT:
					return originalCompilationUnit;
				case Java_PackageImpl.NAMESPACEACCESS_ORIGINALCLASSFILE:
					return originalClassFile;
			}
			return base.eGet(featureID, resolve, coreType);
		}
		
		
		public override void eSet(int featureID, object newValue) {
			switch (featureID) {
				case Java_PackageImpl.NAMESPACEACCESS_COMMENTS:
					comments.Clear();
					comments.AddRange(((List<EObject>)newValue)?.Cast<Comment>());
					return;
				case Java_PackageImpl.NAMESPACEACCESS_ORIGINALCOMPILATIONUNIT:
					originalCompilationUnit = (CompilationUnit) newValue;
					return;
				case Java_PackageImpl.NAMESPACEACCESS_ORIGINALCLASSFILE:
					originalClassFile = (ClassFile) newValue;
					return;
			}
			base.eSet(featureID, newValue);
		}
		
		
	}
}

File: crossecore-typescript/java_/NamespaceAccessBase.ts
import {EClass} from "ecore/EClass";
import {NotificationChain} from "ecore/NotificationChain";
import {ENotificationImpl} from "ecore/ENotificationImpl";
import {NotificationImpl} from "ecore/NotificationImpl";
import {InternalEObject} from "ecore/InternalEObject";
import {BasicEObjectImpl} from "ecore/BasicEObjectImpl";
import {Java_PackageLiterals} from "./Java_PackageLiterals";
import {NamespaceAccess} from "./NamespaceAccess";
import {ASTNodeImpl} from "./ASTNodeImpl";
		
			export class NamespaceAccessBase
			extends ASTNodeImpl
			implements NamespaceAccess
			{

			
				public static eStaticClass:EClass;
				
				protected eStaticClass():EClass{
					
					return NamespaceAccessBase.eStaticClass;
				}
			
			
			
				public eGet_number_boolean_boolean(featureID:number, resolve:boolean, coreType:boolean):any{
					switch (featureID) {
						case Java_PackageLiterals.NAMESPACEACCESS_COMMENTS:
							return this.comments;
						case Java_PackageLiterals.NAMESPACEACCESS_ORIGINALCOMPILATIONUNIT:
							return this.originalCompilationUnit;
						case Java_PackageLiterals.NAMESPACEACCESS_ORIGINALCLASSFILE:
							return this.originalClassFile;
					}
					//return this.eGetFromASTNode(featureID, resolve, coreType);
					return super.eGet(featureID, resolve, coreType);
				}
				
				public eSet_number_any(featureID:number, newValue:any):void {
					switch (featureID) {
					}
					super.eSet_number_any(featureID, newValue);
				}

				
			}
			
